aten/src/THC/THCCachingAllocator.cpp

```diff
@@ -181,9 +195,36 @@ struct THCCachingAllocator
     } else {
       void* ptr;
       size_t alloc_size = small ? kSmallAlloc : size;
-      err = cuda_malloc_retry(device, &ptr, alloc_size);
+      cudaError_t err = cuda_malloc_retry(device, &ptr, alloc_size);
       if (err != cudaSuccess) {
-        return err;
+        if (err == cudaErrorMemoryAllocation) {
+          cudaGetLastError();  // clear CUDA error
+
+          cudaDeviceProp prop;
+          AT_CUDA_CHECK(cudaGetDeviceProperties(&prop, device));
```

Comment:
This is a super slow call, and we have those cached in ATen context, so it would be good to use that.

Reply:
I switched to cudaMemGetInfo to be able to report the free memory as well